src/git.rs

```diff
@@ -79,40 +115,181 @@ impl Git for LiveGit {
             None => Err(Box::new(git2::Error::from_str("No branch name found")))
         }
     }
-}
 
-#[cfg(test)]
-mod test {
-    use super::*;
-    
-    pub struct MockGit;
-    impl Git for MockGit {
-        fn new() -> Result<Self, Box<dyn Error>> {
-            Ok(MockGit)
+    fn log(&self) -> Result<(), Box<dyn Error>> {
+        let mut revwalk = self.repo.revwalk()?;
+        revwalk.push_head()?;
+        for id in revwalk {
+            let id = id?;
```

Comment:
Shadowing id is less readable.  Could we give it another name to make it clearer.